enclave/core/sgx/tracee.c

```diff
@@ -9,28 +9,36 @@
 #include "../tracee.h"
 #include "report.h"
 
+static volatile int _is_enclave_debug_allowed = -1;
+
 // Read an enclave's identity attribute to see to if it was signed as an debug
 // enclave
 bool is_enclave_debug_allowed()
 {
-    bool ret = false;
-    oe_sgx_td_t* td = oe_sgx_get_td();
+    if (_is_enclave_debug_allowed != -1)
+        goto done;
 
+    oe_sgx_td_t* td = oe_sgx_get_td();
     if (td->simulate)
     {
         // enclave in simulate mode is treated as debug_allowed
-        ret = true;
+        _is_enclave_debug_allowed = 1;
     }
     else
     {
         // get a report on the enclave itself for enclave identity information
         sgx_report_t sgx_report;
         oe_result_t result = sgx_create_report(NULL, 0, NULL, 0, &sgx_report);
         if (result != OE_OK)
+        {
+            _is_enclave_debug_allowed = 0;
             goto done;
+        }
 
-        ret = (sgx_report.body.attributes.flags & SGX_FLAGS_DEBUG) != 0;
+        _is_enclave_debug_allowed =
+            (sgx_report.body.attributes.flags & SGX_FLAGS_DEBUG) != 0;
```

Comment:
Better to do the following explicitly, instead of relying on the compiler to set 1 or 0. More readable. Additionally, earlier versions of C did not have a bool type and I'm not really sure what the compiler is allowed to do in such a case.
```suggestion
            (sgx_report.body.attributes.flags & SGX_FLAGS_DEBUG) ? 1 : 0;
```